### PR TITLE
Implementation of Ruby Enumerable Methods

### PR DESCRIPTION
**Pull request by mihrab34**
This is a feature branch of my implementation of Ruby Enumerable Methods

### CODE REVIEW

File: main.rb
Comment (suretrust) [inline main.rb]: - [ ]  Enumerables `my_all?` returns true if the block never returns false or nil
`[1,2,3,4,5].my_all?{ |num| num < 10 }` gives `nil` instead of `true`

- [ ]  Enumerables `my_all?` when no block or argument is given returns true when none of the collection members are false or nil
`[1,2,3,4,5].my_all?` should return `true`

- [ ] Enumerables `my_all?` when a class is passed as an argument returns true if all of the collection is a member of such class
`[1,2,3,4,5].my_all?(Integer)` should return `true`

- [ ] Enumerables `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['saheed', 'oladele', 'suretrust'].my_all?(/d/)` should return `true`

- [ ] Enumerables `my_all?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_all?(3)` should return `false`

File: main.rb
Comment (suretrust) [inline main.rb]: - [ ]  Enumerables `my_any?` returns true if the block never returns false or nil
`[1,2,3,4,5].my_any?{ |num| num < 10 }` gives `nil` instead of `true`

- [ ]  Enumerables `my_any?` when no block or argument is given returns true when none of the collection members are false or nil
`[1,2,3,4,5].my_any?` should return `true`

- [ ] Enumerables `my_any?` when a class is passed as an argument returns true if all of the collection is a member of such class
`[1,2,3,4,5].my_any?(Integer)` should return `true`

- [ ] Enumerables `my_any?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['saheed', 'oladele', 'suretrust'].my_any?(/d/)` should return `true`

- [ ] Enumerables `my_any?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[1,2,3,4,5].my_any?(3)` should return `true`

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @mihrab34,

### Great job thus far, :raised_hands: however, there are some issues with some of the enumerable methods, please see my comments on your code. :eyes: 

**Kindly submit for another code review once these issues have been fixed.** _Happy coding!_ :tada:

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello @mihrab34   🌟

**Great work implementing the changes, however before I can approve your project the following must be done ⬇️**

- The pull request should have a descriptive name and a short description, [refernce](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png).

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Hi @mihrab34 :wave: , please implement the changes suggested by @Redvanisation in order to get approval for the project :wink: 

When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by stratospherique — APPROVED** [review]
### Great work @team :clap: :clap: The project is approved :heavy_check_mark:  Congrats :tada: You may merge :+1: :+1: 

Keep up the good work :muscle: 
_**Ahmed Mahfoudh**_

**Pull request merged**
